dask/array/core.py

```diff
@@ -3187,6 +3224,16 @@ def cov(m, y=None, rowvar=1, bias=0, ddof=None):
         return (dot(X, X.T.conj()) / fact).squeeze()
 
 
+@wraps(np.corrcoef)
+def corrcoef(x, y=None, rowvar=1):
+    c = cov(x, y, rowvar)
+    if c.shape == ():
+        return c/c
+    d = diag(c)
+    d = d.reshape((d.shape[0], 1))
+    return c / sqrt(d * d.T)
```

Comment:
The diagonal is always positive, so that actually would be equivalent. They both result in virtually the same graph though, just a reordering of sqrt and div calls. The way I did it results in slightly fewer tasks in the optimized graph, but the memory usage/execution time is virtually the same either way. Good catch on the `d[None, :]` though, no need to call `reshape` here.